cli/azd/resources/apphost/templates/main.bicept

```diff
@@ -61,7 +61,7 @@ module resources 'resources.bicep' = {
   }
 }
 {{ range $name, $module := .BicepModules }}
-module {{$name}} '{{ fixBackSlash $module.Path }}' = {
+module {{dashToUnderscore $name}} '{{ fixBackSlash $module.Path }}' = {
```

Comment:
Did we consider using [BicepName](https://github.com/Azure/azure-dev/blob/88a51a039d6af636deacffaf1735b94a165281ef/cli/azd/internal/scaffold/funcs.go#L14)?

Reply:
Yes, but I think we have a hidden bug with `bicepName()` as that function removes the `-` instead of replacing it for something else. That means that adding `foo-01` and `foo01` crashes (resources would be unique on AppHost but would become the same name within bicep)

Reply:
As a user, I don't think I would be too upset about not being able to define `foo-01` and `foo01` as two separate distinct resources. I think I'd care more about "is the bicep produced more idiomatic?" (at least when the bicep was generated on-disk and wasn't an implementation detail), which camel-case seems to be more preferred.

Reply:
> which camel-case seems to be more preferred.

Underscore is a valid char for bicep identifiers, we should use it regardless of what's the idiomatic.  

I think anyone who prefers camelCase would have written `someValue` in the appHost instead of `some-value`.